stdlib/jsonutils/src/main/java/org/ballerinalang/stdlib/jsonutils/JSONToXMLConverter.java

```diff
@@ -0,0 +1,218 @@
+     */
+    @SuppressWarnings("rawtypes")
+    private static List<XMLValue> traverseTree(Object json, String attributePrefix, String arrayEntryTag) {
+        List<XMLValue> xmlArray = new ArrayList<>();
+        if (!(json instanceof RefValue)) {
+            XMLValue xml = XMLFactory.parse(json.toString());
+            xmlArray.add(xml);
+        } else {
+            traverseJsonNode(json, null, null, xmlArray, attributePrefix, arrayEntryTag);
+        }
+        return xmlArray;
+    }
+
+    /**
+     * Traverse a JSON node ad produces the corresponding xml items.
+     *
+     * @param json               to be traversed
+     * @param nodeName           name of the current traversing node
+     * @param parentElement      parent element of the current node
+     * @param omElementArrayList List of XML items generated
+     * @param attributePrefix    String prefix used for attributes
+     * @param arrayEntryTag      String used as the tag in the arrays
+     * @return List of XML items generated during the traversal.
+     */
+    @SuppressWarnings("rawtypes")
+    private static OMElement traverseJsonNode(Object json, String nodeName,
+            OMElement parentElement, List<XMLValue> omElementArrayList, String attributePrefix, String arrayEntryTag) {
+        OMElement currentRoot = null;
+        if (nodeName != null) {
+            // Extract attributes and set to the immediate parent.
+            if (nodeName.startsWith(attributePrefix)) {
+                if (json instanceof RefValue) {
+                    throw new BallerinaException("attribute cannot be an object or array");
+                }
+                if (parentElement != null) {
+                    String attributeKey = nodeName.substring(1);
+                    // Validate whether the attribute name is an XML supported qualified name, according to the XML
+                    // recommendation.
+                    XMLValidator.validateXMLName(attributeKey);
+
+                    parentElement.addAttribute(attributeKey, json.toString(), null);
+                }
+                return parentElement;
+            }
+
+            // Validate whether the tag name is an XML supported qualified name, according to the XML recommendation.
+            XMLValidator.validateXMLName(nodeName);
+
+            currentRoot = OM_FACTORY.createOMElement(nodeName, null);
+        }
+
+        if (json == null) {
+            OMNamespace xsiNameSpace = OM_FACTORY.createOMNamespace(XSI_NAMESPACE, XSI_PREFIX);
+            currentRoot.addAttribute(NIL, "true", xsiNameSpace);
+        } else {
+            LinkedHashMap<String, Object> map;
+
+            BType type = TypeChecker.getType(json);
+            switch (type.getTag()) {
+
+            case TypeTags.MAP_TAG:
+                if (((BMapType) type).getConstrainedType().getTag() != TypeTags.JSON_TAG) {
+                    throw new BallerinaException("error in converting map<non-json> to xml");
```

Comment:
Throw ErrorValue instead of BallerinaException in all the places.

Reply:
Added